pop-cli-for-appchains/guides/test-runtime-upgrades.md

```diff
@@ -0,0 +1,157 @@
+│  wss://rpc1.paseo.popnetwork.xyz
+│
+◆  Enter the block hash (optional):
+│  0x1234567890abcdef1234567890abcdef
+└
+```
+
+You'll be asked to enter the URI of a live node and optionally provide a block hash. If a [block hash](https://paritytech.github.io/polkadot-sdk/master/sp_runtime/traits/trait.HashOutput.html) is given, the state will be executed at the specified block hash on the provided network.
+
+To run the migrations on top of live state manually:
+
+```bash
+pop test on-runtime-upgrade \
+    --runtime=<path_to_runtime_binary> \
+    live \
+    --uri=wss://rpc1.paseo.popnetwork.xyz \
+    --at=0x1234567890abcdef1234567890abcdef
+```
+
+**_Note_**: The specified runtime and the remote node's runtime must have the same name and version. If not, the migration will fail. You can add the flag `--disable-spec-version-check` and `--disable-spec-name-check` to bypass the checks.
+
+```bash
+pop test on-runtime-upgrade \
+    --runtime=<path_to_runtime_binary> \
+    --disable-spec-version-check \
+    --disable-spec-name-check \
+    live \
+    --uri=wss://rpc1.paseo.popnetwork.xyz \
+    --at=0x1234567890abcdef1234567890abcdef
+```
+
+After that, you can select select the upgrade checks to perform:
+
+```bash
+◆  Select upgrade checks to perform:
+│  ○ none
+│  ● all (Run the `try_state`, `pre_upgrade` and `post_upgrade` checks)
+│  ○ try-state
+│  ○ pre-and-post
+└
+```
+
+### Test migrations with a snapshot file
+
+A second approach to test migrations is with a snapshot file. First, you need to create a snapshot file of a live network. You can do this by running the following command:
+
+```bash
+pop test create-snapshot
+```
+
+The interactive interface to prompts for the live URI and the path of the snapshot file:
+
+```bash
+┌   Pop CLI : Creating a snapshot file
+│
+▲  NOTE: `create-snapshot` only works with the remote node. No runtime required.
+│
+◇  Enter the URI of the remote node:
+│  wss://rpc1.paseo.popnetwork.xyz
+│
+◇  Enter the path to write the snapshot to (optional):
+│  If not provided `<spec-name>-<spec-version>@<block-hash>.snap` will be used.
+│  example.snap
+```
+
+To skip the interactive prompt, use the `--uri` and `--path` flags:
+
+```bash
+pop test create-snapshot --uri wss://rpc1.paseo.popnetwork.xyz ./example.snap
+```
+
+If the path of snapshot file is not provided, the default name following a format `<spec-name>-<spec-version>@<block-hash>.snap` will be used. Note that the remote node must be built with `--try-runtime` feature enabled.
+
+Assume there is a snapshot file created with the name `example.snap`:
+
+```bash
+◇  Enter path to your snapshot file?
+│  Snapshot file can be generated using `pop test create-snapshot` command.
+│  ./example.snap
+```
+
+To run migrations with a snapshot manually:
+
+```bash
+pop test on-runtime-upgrade \
+    --runtime=<path_to_runtime_binary> \
+    --blocktime=6000 \
+    --checks=all \
+    snap \
+    --path=./example.snap
+```
+
+#### Learning Resources
```

Comment:
Maybe a link to `try-runtime-cli` https://github.com/paritytech/try-runtime-cli ? and also to the docs: https://paritytech.github.io/try-runtime-cli/try_runtime/